js/ui/tree_list/ui.tree_list.selection.js

```diff
@@ -416,27 +430,39 @@ treeListCore.registerModule("selection", extend(true, {}, selectionModule, {
                     return result;
                 },
 
+                /**
+                * @name dxTreeListMethods_getSelectedRowKeys
+                * @publicName getSelectedRowKeys(leavesOnly)
+                * @param1 leavesOnly:boolean
+                * @return Array<any>
+                * @deprecated
+                */
+
                 /**
                 * @name dxTreeListMethods_getSelectedRowKeys
                 * @publicName getSelectedRowKeys(mode)
-                * @param1 mode:string|boolean
+                * @param1 mode:string
                 * @return Array<any>
                 */
                 getSelectedRowKeys: function(mode) {
                     var that = this,
                         dataController = that._dataController,
                         selectedRowKeys = that.callBase.apply(that, arguments) || [];
 
-                    if(this.isRecursiveSelection() && dataController) {
-                        if(that._isModeLeavesOnly(mode)) {
-                            selectedRowKeys = dataController.getNodeLeafKeys(selectedRowKeys, function(childNode, nodes) {
-                                return !childNode.hasChildren && that.isRowSelected(childNode.key);
-                            });
+                    mode = mode || "excludeRecursive";
+
+                    if(dataController) {
+                        if(mode === true) {
+                            errors.log("W0002", "dxTreeList", "getSelectedRowKeys(leavesOnly)", "18.1", "Use the 'getSelectedRowKeys(mode)' method with a string parameter instead");
                         }
 
-                        if(mode === "all") {
+                        if(this.isRecursiveSelection() && mode !== "excludeRecursive") {
```

Comment:
I think it's better to check on 'includeRecursive': `mode === "includeRecursive"`

Reply:
After refix this string can look like this or ```mode === "includeRecursive" || mode === "leavesOnly"```